packages/client/src/index.ts

```diff
@@ -78,6 +103,6 @@ export interface DelegatedRoutingV1HttpApiClient {
 /**
  * Create and return a client to use with a Routing V1 HTTP API server
  */
-export function createDelegatedRoutingV1HttpApiClient (url: URL, init: DelegatedRoutingV1HttpApiClientInit = {}): DelegatedRoutingV1HttpApiClient {
-  return new DefaultDelegatedRoutingV1HttpApiClient(url, init)
+export function createDelegatedRoutingV1HttpApiClient (url: URL | string, init: DelegatedRoutingV1HttpApiClientInit = {}): DelegatedRoutingV1HttpApiClient {
+  return new DefaultDelegatedRoutingV1HttpApiClient(new URL(url), init)
```

Comment:
+1 for supporting strings

Reply:
In general if we can push validation of an argument type out of a consumer it's a good thing.  Otherwise this can snowball quickly and we have to guess what the contents of a string are, usually at several points in a module if we have multiple entry points.

We suffered from this previously with "what's a CID?" is it "QmFoo", is it "/ipfs/QMFoo", etc.  Pushing all that into the `CID` class vastly simplified the codebase.

Of course for a user it's nicer to be able to pass `'http://example.org'` instead of `new URL('http://example.org')` so it's probably worth the tradeoff for the `create...` factory functions.

Reply:
this should be a straightforward one where it will fail quickly if an invalid string is passed.